### ISSUE TITLE
unable to use class constants as translation keys

### ISSUE THREAD

**Issue by slano** [kirschbaum-development/laravel-translations-loader#5]
Hi,

I'd like to use this loader to prepare language strings for vue, but I'm using BenSampo/laravel-enum package, and my translation arrays for enums looks like this:

```
<?php

use App\Enums\OfferStatus;

return [
    OfferStatus::class => [
        OfferStatus::New => 'new',
        OfferStatus::Closed => 'closed',
    ],
];
```

laravel-translation-loader is unfortunately unable to parse such php file. I get the following error:

```
 error  in ./node_modules/@kirschbaum-development/laravel-translations-loader/all.js

Module build failed: Error: Unexpected PHP key: "staticlookup", details: {"kind":"staticlookup","loc":{"source":null,"start":{"line":3,"column":28,"offset":32},"end":{"line":3,"column":35,"offset":39}},"what":{"kind":"identifier","loc":{"source":null,"start":{"line":3,"column":4,"offset":8},"end":{"line":3,"column":28,"offset":32}},"resolution":"uqn","name":"OfferStatus"},"offset":{"kind":"constref","loc":{"source":null,"start":{"line":3,"column":30,"offset":34},"end":{"line":3,"column":35,"offset":39}},"name":"class"}}
...
```

would it be possible to implement lookup for class constants?

A workaround was to rewrite translation array like this:

```
<?php

return [
    'App\Enums\OfferStatus' => [
        1 => 'new',
        2 => 'closed',
    ],
];
```

but that's not very elegant...

**Comment by luisdalmolin** [comment]
Hey - Yeah that's definitely not the more elegant solution. Behind the scenes, we use https://www.npmjs.com/package/php-array-parser to parse the Laravel translation files. The ideal solution here would be for this package to provide support for this kind of stuff, but this doesn't seem something very simple. I may take a deeper look at this issue again whenever I have some free time.

**Comment by luisdalmolin** [comment]
@slano if you have any suggestions on how to solve this we are open to PRs. Otherwise, I'll take a look as soon as I can.

**Comment by slano** [comment]
I guess a major functionality would have to be implemented into php-array-parser, I'll just keep using a workaround for now, it gets the job done.

**Comment by dnwjn** [comment]
Unfortunately this issue has still not been resolved (using `v1.3.0`). It also seems like [php-array-parser](https://github.com/rmariuzzo/php-array-parser) has been abandoned.

**Comment by luisdalmolin** [comment]
Yes, this is still a thing. To make this work we basically would have to run `php file.php` and process the result. It may not be so bad, but definitely introduces a few potential problems. I may look into this when I have some time, but feel free to send PRs for this. This would not need any updates on the php-array-parser side, it would be work done before that.
